src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -582,74 +637,74 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         auto lifetime = get_strong();
 
-        FileOpenPicker picker;
-
-        _State.WindowRoot().TryPropagateHostingWindow(picker); // if we don't do this, there's no HWND for it to attach to
-        picker.ViewMode(PickerViewMode::Thumbnail);
-        picker.SuggestedStartLocation(PickerLocationId::PicturesLibrary);
-
-        // Converted into a BitmapImage. This list of supported image file formats is from BitmapImage documentation
-        // https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Media.Imaging.BitmapImage?view=winrt-19041#remarks
-        picker.FileTypeFilter().ReplaceAll({ L".jpg", L".jpeg", L".png", L".bmp", L".gif", L".tiff", L".ico" });
-
-        StorageFile file = co_await picker.PickSingleFileAsync();
-        if (file != nullptr)
+        auto file = co_await OpenImagePicker();
```

Comment:
Do we need to hop to the background thread before or during this `co_await`? Or does the IFileDialog do that automagically?

Reply:
It's actually the co_await that splits off to the right thread! However: we actually do want to block the UI thread- the dialog is naturally a UI thing, and we don't want the user navigating to another profile page while the dialog is up. Yes, that sucks, but I don't believe this is a difference from Pickers.

Reply:
Oh, wait really? I always assumed that until you `resume_*` then you're still on the main thread.

I guess blocking the UI thread is in fact the right behavior, I guess I just didn't want the window to do the "Window unresponsive" thing, but I guess Windows knows how to deal with that already.